test_crates/repr_c_plain_struct_fields_reordered/new/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// Public repr(C) struct with reordered fields - should trigger warning
+#[repr(C)]
+pub struct PublicStruct {
+    pub b: u16, // moved
+    pub c: u32, // moved
+    pub a: u8,  // moved
+}
+
+// Private repr(C) struct with reordered fields - should not trigger
+#[repr(C)]
+struct PrivateStruct {
+    pub b: u16,
+    pub c: u32,
+    pub a: u8,
+}
+
+// Public non-repr(C) struct with reordered fields - should not trigger
+pub struct RegularStruct {
+    pub b: u16,
+    pub c: u32,
+    pub a: u8,
+}
+
+// Public repr(C) struct with doc(hidden) - should not trigger
+#[doc(hidden)]
+#[repr(C)]
+pub struct DocHiddenStruct {
+    pub b: u16,
+    pub c: u32,
+    pub a: u8,
+}
+
+// Public repr(C) struct with mixed visibility fields
+#[repr(C)]
+pub struct MixedVisibilityStruct {
+    b: u16,     // private field moved
```

Comment:
We probably don't want to trigger the lint on the private field itself, just on the `pub` ones. If a field isn't `pub`, it arguably shouldn't be accessed in a binary fashion just as much as it shouldn't be accessed via the API.

So for example, the following shouldn't trigger the lint at all:
```rust
#[repr(C)]
pub struct PrivateReordered {
    pub a: i64,
    b: i64,  // this field
    c: i64,  // and this field swap places
    pub d: i64,
}
```

Reply:
so if field `b` and `a` swap places, we won't trigger the lint for `b` right just `a` ?

Reply:
yeah I think I covered both, new added tests are not triggering the lint at all, and one original test has private field and pub field change position